### ISSUE TITLE
Curly braces in rewrite result in broken formatting in nginx config

### ISSUE THREAD

**Issue by lukasbableck** [Froxlor/Froxlor#1171]
**Describe the bug**
If a rewrite in the vHost settings for a domain contains curly braces Froxlor seems to add whitespace around those braces, obviously breaking the rewrite.

This
<img width="932" alt="Bildschirmfoto 2023-08-22 um 08 20 45" src="https://github.com/Froxlor/Froxlor/assets/42083846/9a9309b8-e11a-4354-ba9f-c9244c41b571">

results in
<img width="322" alt="image" src="https://github.com/Froxlor/Froxlor/assets/42083846/3989c659-0982-4292-9943-632cf04136b9">

**System information**
* Froxlor version: 2.0.21 (also happens in a7198f58ce9787c7b67210d6d2eec8f95ceea775)
* Web server: nginx
* OS/Version: Debian 11

**To Reproduce**
1. Go to any domain
2. Add for example 'rewrite "^/\d{4}/\d{2}/\d{2}(/.*)$" $1 permanent;' to it's vhost settings
3. Check generated vhost config and see incorrectly formatted rewrite

**Expected behavior**
No whitespaces before and after curly braces if they are part of a string

**Comment by d00p** [comment]
This requires some deep work in the nginx-custom-config merge process. The cause it not the replacers but the curly brackets itself being opening and closing blocks. We'll see what we can do

**Comment by d00p** [comment]
Could you test the following patch and check whether it works as intended?

```diff
diff --git a/lib/Froxlor/Cron/Http/Nginx.php b/lib/Froxlor/Cron/Http/Nginx.php
index 76ad0471..c0cf4749 100644
--- a/lib/Froxlor/Cron/Http/Nginx.php
+++ b/lib/Froxlor/Cron/Http/Nginx.php
@@ -883,13 +883,7 @@ class Nginx extends HttpConfigBase
                // remove comments
                $vhost = implode("\n", preg_replace('/^(\s+)?#(.*)$/', '', explode("\n", $vhost)));
                // Break blocks into lines
-               $vhost = str_replace([
-                       "{",
-                       "}"
-               ], [
-                       " {\n",
-                       "\n}"
-               ], $vhost);
+               $vhost = preg_replace("/^(\s+)location(.+)\{(.+)\}$/misU", "location $2 {\n $3 \n}", $vhost);
                // Break into array items
                $vhost = explode("\n", preg_replace('/[ \t]+/', ' ', trim(preg_replace('/\t+/', '', $vhost))));
                // Remove empty lines
```

**Comment by d00p** [comment]
fixed in 9d0e463906c4f2df3ac39aafdca2737260d2e433
